### DISCUSSION TITLE
CloudwatchAgent on runners

### DISCUSSION THREAD

**Discussion by npeyrussie** [npalm/terraform-aws-gitlab-runner#347] [Q&A]
Hello,

I am looking for a way to install CloudwatchAgent on the runners.
I seems that `userdata_pre_install` and `userdata_post_install` only apply to the agent and not to the runners.

What type of detailed monitoring metrics are added thanks to `runner_instance_enable_monitoring` option ?

I need to retrieve memory metrics in cloudwatch and need to have cloudwatch agent installed on runners.
Is there a way to achieve this ?

**Comment by npeyrussie** [comment]
I eventually found my way by generating a script file in `userdata_post_install` and then using its path in 
`docker_machine_options` as option `amazonec2-userdata` value.

I also added 
`arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy` to `docker_machine_iam_policy_arns` variable

**Reply by kayman-mk** [reply]
#408 describes this in detail.
